### PR TITLE
Added @material-ui/lab & fix main entry points

### PR DESCRIPTION
**Pull request by jollytoad**

### CODE REVIEW

File: overrides.json [hunk 1]
```diff
@@ -35,6 +35,25 @@
       }
     }
   },
+  "@material-ui/lab": {
+    "4": {
+      "exports": {
+        ".": "./index.js",
```
Comment (guybedford) [inline overrides.json:41]: Should this also be something like:

```suggestion
        ".": {
          "require": "./index.js",
          "import": "./esm/index.js"
        }
```

File: overrides.json [hunk 2]
```diff
@@ -94,8 +101,7 @@
   "@material-ui/utils": {
     "4": {
       "exports": {
-        ".": "./index.js",
-        "./index.js": {
+        ".": {
           "require": "./index.js",
           "import": "./esm/index.js"
         },
```
Comment (guybedford) [inline overrides.json:107]:
```suggestion
        },
        "./index.js": "./index.js",
```

File: overrides.json [hunk 3]
```diff
@@ -76,8 +84,7 @@
   "@material-ui/system": {
     "4": {
       "exports": {
-        ".": "./index.js",
-        "./index.js": {
+        ".": {
           "require": "./index.js",
           "import": "./esm/index.js"
         },
```
Comment (guybedford) [inline overrides.json:90]:
```suggestion
        },
        "./index.js": "./index.js",
```

File: overrides.json [hunk 4]
```diff
@@ -57,7 +62,10 @@
...
     "4": {
       "exports": {
-        ".": "./index.js",
+        ".": {
+          "require": "./index.js",
+          "import": "./esm/index.js"
+        },
         "./index.js": {
           "require": "./index.js",
           "import": "./esm/index.js"
```
Comment (guybedford) [inline overrides.json:71]: I can't suggest it here, but this should just be `"./index.js": "./index.js"`.

File: overrides.json [hunk 5]
```diff
@@ -38,7 +40,10 @@
   "@material-ui/lab": {
     "4": {
       "exports": {
-        ".": "./index.js",
+        ".": {
+          "require": "./index.js",
+          "import": "./esm/index.js"
+        },
         "./index.js": {
```
Comment (guybedford) [inline overrides.json:47]: `"./index.js": "./index.js"`.

File: overrides.json [hunk 6]
```diff
@@ -19,7 +18,10 @@
   "@material-ui/core": {
     "4": {
       "exports": {
-        ".": "./index.js",
+        ".": {
+          "require": "./index.js",
+          "import": "./esm/index.js"
+        },
         "./index.js": {
```
Comment (guybedford) [inline overrides.json:25]: `"./index.js": "./index.js"`.

File: overrides.json [hunk 7]
```diff
@@ -2,8 +2,7 @@
   "@material-ui/icons": {
     "4": {
       "exports": {
-        ".": "./index.js",
-        "./index.js": {
+        ".": {
           "require": "./index.js",
           "import": "./esm/index.js"
         },
```
Comment (guybedford) [inline overrides.json:8]:
```suggestion
        },
        "./index.js": "./index.js",
```

**Review by guybedford — CHANGES_REQUESTED** [review]

File: overrides.json (hunk 2 above)
Comment (jollytoad) [reply overrides.json:107]: ok, so if index.js is explicitly required then we explicitly provide that rather than the esm override?

Comment (guybedford) [reply overrides.json:100]: Yes because that's the expectation since without exports that's what they'd get before.

**Pull request merged**